kb.py

```
@@ -0,0 +1,584 @@
+                flavor_assets = data[2].get('flavorAssets', [])
+                if isinstance(flavor_assets, list) and flavor_assets:
+                    # Heuristic: pick the first flavor asset that is not 'audio*' or 'image*' if possible
+                    # and hope it's a video.
+                    for asset in flavor_assets:
+                        tags = asset.get('tags', '')
+                        if 'audio' not in tags and 'image' not in tags and 'caption' not in tags: # try to avoid non-video
+                            current_flavor_id = asset.get('id')
+                            file_ext = asset.get('fileExt', 'mp4')
+                            break
+                    if not current_flavor_id and flavor_assets: # If still nothing, just take the first one
+                         current_flavor_id = flavor_assets[0].get('id')
```

Flake8 (7.2.0)

[error] indentation is not a multiple of 4

(E111)